src/jats/jats-exporter.ts

```diff
@@ -1749,57 +1752,29 @@ export class JATSExporter {
           }
         })
       }
-      const noteIDs: string[] = []
-      for (const contributor of [...authorContributors, ...otherContributors]) {
-        if (contributor.footnote) {
-          const ids = contributor.footnote.map((note) => {
-            return note.noteID
-          })
-          noteIDs.push(...ids)
-        }
-        if (contributor.corresp) {
-          const ids = contributor.corresp.map((corresp) => {
-            return corresp.correspID
-          })
-          noteIDs.push(...ids)
-        }
-      }
-      const footnotes: Footnote[] = []
-      footnotes.push(
-        ...this.models.filter(hasObjectType<Footnote>(ObjectTypes.Footnote))
+      const authorNotes = this.models.find(
+        hasObjectType<AuthorNotes>(ObjectTypes.AuthorNotes)
       )
-
-      const correspodings: Corresponding[] = []
-      correspodings.push(
-        ...this.models.filter(
-          hasObjectType<Corresponding>(ObjectTypes.Corresponding)
-        )
-      )
-
-      if (footnotes || correspodings) {
+      if (authorNotes) {
         const authorNotesEl = this.document.createElement('author-notes')
-        const usedFootnotes = footnotes.filter((footnote) => {
-          return noteIDs.includes(footnote._id)
-        })
-        const usedCorrespodings = correspodings.filter((corresp) => {
-          return noteIDs.includes(corresp._id)
-        })
-        usedFootnotes.forEach((footnote) => {
-          const authorFootNote = this.document.createElement('fn')
-          authorFootNote.setAttribute('id', normalizeID(footnote._id))
-          authorFootNote.innerHTML = footnote.contents
-          authorNotesEl.appendChild(authorFootNote)
-        })
-        usedCorrespodings.forEach((corresponding) => {
-          const correspondingEl = this.document.createElement('corresp')
-          correspondingEl.setAttribute('id', normalizeID(corresponding._id))
-          if (corresponding.label) {
-            const labelEl = this.document.createElement('label')
-            labelEl.textContent = corresponding.label
-            correspondingEl.appendChild(labelEl)
+        authorNotes.containedObjectIDs.forEach((id) => {
+          const model = this.modelMap.get(id)
+          if (!model) {
+            return
+          }
+          if (id.startsWith('MPParagraphElement')) {
+            this.appendParagraphToElement(
+              model as ParagraphElement,
+              authorNotesEl
+            )
+          } else if (id.startsWith('MPCorresponding')) {
+            this.appendCorrespondingToElement(
+              model as Corresponding,
+              authorNotesEl
+            )
+          } else if (id.startsWith('MPFootnote')) {
```

Comment:
The corresp is not part of the author_notes node, is it?

Reply:
I kept it in case we do add corresp in the future, but i can remove it.